File: kafka/client-node/consumer.js
import { Kafka } from 'kafkajs'

const kafka = new Kafka({
    clientId: 'consumer-node',
    brokers: ['localhost:10200', 'localhost:10201', 'localhost:10202'],
})

const consumer = kafka.consumer({ groupId: 'consumer-foo' })
await consumer.connect()

process.on('SIGINT', async () => {
    console.log('Caught interrupt signal')
    await consumer.disconnect()
    process.exit()
})

await consumer.subscribe({ topic: 'foo', fromBeginning: true })
consumer.run({
    eachMessage: async ({ topic, partition, message }) =>
        console.log(`${topic}-${partition} key: ${message.key} value: ${message.value}`),
})

File: kafka/client-python/consumer.py
import datetime
import random
import signal
import sys
import time
from typing import cast

import kafka

consumer = kafka.KafkaConsumer(
    "foo",
    client_id="consumer-python",
    group_id="consumer-foo",
    bootstrap_servers=["localhost:9080", "localhost:9081", "localhost:9082"],
)


def signal_handler(sig: int, frame: object) -> None:
    print("Caught interrupt signal")
    consumer.close(5)
    sys.exit(0)


signal.signal(signal.SIGINT, signal_handler)

for message in consumer:
    print(
        f'{message.topic}-{message.partition} key: {message.key and cast(bytes, message.key).decode("utf-8")} key: {message.value and cast(bytes, message.value).decode("utf-8")}'
    )
